api/v1/databaseengine_types.go

```diff
@@ -39,13 +39,15 @@ type (
 
 // DatabaseEngineSpec is a spec for a database engine.
 type DatabaseEngineSpec struct {
-	Type EngineType `json:"type"`
+	Type            EngineType `json:"type"`
+	EnabledVersions Versions   `json:"enabledVersions,omitempty"`
```

Comment:
But we're not gonna define the whole `Versions` object here in the spec, right? At least I believe the component's `Status` doesn't make sense by creation.

Reply:
This is a good point, the user shouldn't have to specify the component at all, maybe just the keys which are the actual versions.
Something like
```
type Versions struct {
	Engine []string
	Backup []string
	Proxy  map[string][]string
	Tools  map[string][]string
```

Reply:
Yeah. I agree. Thanks. Anyway I'll try to play with it and it will be clear what to do with the spec

Reply:
Actually we need to specify only database engine versions because we do not manage versions of other components. I changed it. I'll come up with better naming